### PR TITLE
fix: set default region for AWS cli

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It passes the AWS region in which the e2e testing is creating VMs, so that the CLI is able to connect and remove the zombie instances.

## Why is it important?
We'd like to remove running instances in a weekly basis, but we saw this error:

```
[2022-01-25T10:15:12.001Z] + aws ec2 describe-instances --filters Name=tag:ReaperMark,Values=e2e-testing-vm --query Reservations[].Instances[].InstanceId --output text
[2022-01-25T10:15:12.945Z]
[2022-01-25T10:15:12.945Z] You must specify a region. You can also configure your region by running "aws configure".
[2022-01-25T10:15:12.945Z] + aws ec2 terminate-instances --instance-ids
[2022-01-25T10:15:13.895Z]
[2022-01-25T10:15:13.895Z] You must specify a region. You can also configure your region by running "aws configure".
script returned exit code 253
```

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

### CODE REVIEW

File: .ci/aws-instances-reaper.groovy [hunk 1]
```diff
@@ -10,6 +10,7 @@ pipeline {
     PIPELINE_LOG_LEVEL = 'INFO'
     JOB_GIT_CREDENTIALS = "f6c7695a-671e-4f4f-a331-acdce44ff9ba"
     AWS_PROVISIONER_SECRET = 'secret/observability-team/ci/elastic-observability-aws-account-auth'
+    AWS_EC2_INSTANCES_REGION= 'us-east-2'
```
Comment (v1v) [inline .ci/aws-instances-reaper.groovy:13]: Will the below work? Or the `withAWSEnv` configures the default region with the CSV file?

```suggestion
    AWS_EC2_INSTANCES_REGION = 'us-east-2'
    AWS_DEFAULT_REGION = 'us-east-2'
```

Comment (mdelapenya) [reply .ci/aws-instances-reaper.groovy:13]: I've not seen the step to configure the region (https://github.com/elastic/apm-pipeline-library/blob/master/vars/withAWSEnv.groovy).

And you are right, I guess simply declaring the AWS_DEFAULT_REGION variable would be enough, without having to re-export it using `withEnv`

File: .ci/aws-instances-reaper.groovy [hunk 2]
```diff
@@ -30,7 +31,9 @@ pipeline {
     stage('Reap AWS instances'){
       steps {
         withAWSEnv(secret: "${env.AWS_PROVISIONER_SECRET}", forceInstallation: true) {
-          sh("aws ec2 terminate-instances --instance-ids `aws ec2 describe-instances --filters Name=tag:${env.AWS_EC2_INSTANCES_TAG_NAME},Values=${env.AWS_EC2_INSTANCES_TAG_VALUE} --query Reservations[].Instances[].InstanceId --output text`")
+          withEnv(["AWS_DEFAULT_REGION=${AWS_EC2_INSTANCES_REGION}"]) {
+            sh("aws ec2 terminate-instances --instance-ids `aws ec2 describe-instances --filters Name=tag:${env.AWS_EC2_INSTANCES_TAG_NAME},Values=${env.AWS_EC2_INSTANCES_TAG_VALUE} --query Reservations[].Instances[].InstanceId --output text`")
+          }
```
Comment (mdelapenya) [inline .ci/aws-instances-reaper.groovy:36]:
```suggestion
          sh("aws ec2 terminate-instances --instance-ids `aws ec2 describe-instances --filters Name=tag:${env.AWS_EC2_INSTANCES_TAG_NAME},Values=${env.AWS_EC2_INSTANCES_TAG_VALUE} --query Reservations[].Instances[].InstanceId --output text`")
```

File: .ci/aws-instances-reaper.groovy (hunk 1 above)
Comment (v1v) [reply .ci/aws-instances-reaper.groovy]: > I've not seen the step to configure the region

Sorry, I meant if the authetnication with the csv file somehow injects the value from the `csv` data structure, I just verified the csv data structure does not contain any default region:

```
 Import CSV credentials generated from the AWS web console. A CSV file is imported with the profile name matching the IAM user name. The CSV file must contain the following headers.

User Name

Access key ID

Secret access key
```

https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html

**Review by v1v — APPROVED** [review]

**Pull request merged**